Build failing on macOS 14.8.3 (Sonoma)

### Context

I've the latest Xcode cli tools, as well as latest supported Xcode installed.  This has been an issue for a while (building failed when I last tried about 3 months ago)

```
% brew install --HEAD LouisBrunner/valgrind/valgrind

==> Auto-updating Homebrew...
Adjust how often this is run with `$HOMEBREW_AUTO_UPDATE_SECS` or disable with
`$HOMEBREW_NO_AUTO_UPDATE=1`. Hide these hints with `$HOMEBREW_NO_ENV_HINTS=1` (see `man brew`).
==> Fetching downloads for: valgrind
✔︎ Formula valgrind (HEAD-176d087)
==> Installing valgrind from louisbrunner/valgrind
==> ./autogen.sh
==> ./configure
==> make
Last 15 lines from /Users/greg/Library/Logs/Homebrew/valgrind/03.make.log:
clang -DHAVE_CONFIG_H -I. -I..  -I.. -I../include -I../include -I../VEX/pub -I../VEX/pub -DVGA_arm64=1 -DVGO_darwin=1 -DVGP_arm64_darwin=1 -DVGPV_arm64_darwin_vanilla=1    -arch arm64e -O2 -g -Wall -Wmissing-prototypes -Wshadow -Wpointer-arith -Wstrict-prototypes -Wmissing-declarations -Wno-unused-result -Wcast-align -Wcast-qual -Wwrite-strings -Wempty-body -Wformat -Wformat-security -Wignored-qualifiers -Wenum-conversion -finline-functions -fno-stack-protector -fno-strict-aliasing -fno-builtin -Wno-cast-align -Wno-self-assign -Wno-tautological-compare  -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include -mmacos-version-min=11.0 -dynamic -O -g -fno-omit-frame-pointer -fno-strict-aliasing -fpic -fPIC -fno-builtin  -O2  -c -o vgpreload_memcheck_arm64_darwin_so-mc_replace_strmem.o `test -f 'mc_replace_strmem.c' || echo './'`mc_replace_strmem.c
clang  -arch arm64e -O2 -g -Wall -Wmissing-prototypes -Wshadow -Wpointer-arith -Wstrict-prototypes -Wmissing-declarations -Wno-unused-result -Wcast-align -Wcast-qual -Wwrite-strings -Wempty-body -Wformat -Wformat-security -Wignored-qualifiers -Wenum-conversion -finline-functions -fno-stack-protector -fno-strict-aliasing -fno-builtin -Wno-cast-align -Wno-self-assign -Wno-tautological-compare  -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include -mmacos-version-min=11.0 -dynamic -O -g -fno-omit-frame-pointer -fno-strict-aliasing -fpic -fPIC -fno-builtin  -O2  -dynamic -dynamiclib -all_load ../coregrind/libreplacemalloc_toolpreload-arm64-darwin.a  -o vgpreload_memcheck-arm64-darwin.so vgpreload_memcheck_arm64_darwin_so-mc_replace_strmem.o  
../coregrind/link_tool_exe_darwin 0x158000000 clang     -o memcheck-arm64-darwin   -arch arm64 -O2 -g -Wall -Wmissing-prototypes -Wshadow -Wpointer-arith -Wstrict-prototypes -Wmissing-declarations -Wno-unused-result -Wcast-align -Wcast-qual -Wwrite-strings -Wempty-body -Wformat -Wformat-security -Wignored-qualifiers -Wenum-conversion -finline-functions -fno-stack-protector -fno-strict-aliasing -fno-builtin -Wno-cast-align -Wno-self-assign -Wno-tautological-compare  -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include -mmacos-version-min=11.0 -O2 -nodefaultlibs -nostartfiles -Wl,-u,__start -Wl,-e,__start -arch arm64 memcheck_arm64_darwin-mc_leakcheck.o memcheck_arm64_darwin-mc_malloc_wrappers.o memcheck_arm64_darwin-mc_main.o memcheck_arm64_darwin-mc_main_asm.o memcheck_arm64_darwin-mc_translate.o memcheck_arm64_darwin-mc_machine.o memcheck_arm64_darwin-mc_errors.o ../coregrind/libcoregrind-arm64-darwin.a ../VEX/libvex-arm64-darwin.a -lgcc   
link_tool_exe_darwin: /usr/bin/ld -Z -L../coregrind -lmySystem -lmydyld -segaddr __TEXT 0x158000000 -headerpad 256 -new_linker -arch arm64 -macos_version_min 11.0 -o memcheck-arm64-darwin -u __start -e __start -stack_size 0x800000 memcheck_arm64_darwin-mc_leakcheck.o memcheck_arm64_darwin-mc_malloc_wrappers.o memcheck_arm64_darwin-mc_main.o memcheck_arm64_darwin-mc_main_asm.o memcheck_arm64_darwin-mc_translate.o memcheck_arm64_darwin-mc_machine.o memcheck_arm64_darwin-mc_errors.o ../coregrind/libcoregrind-arm64-darwin.a ../VEX/libvex-arm64-darwin.a
ld: warning: -new_linker is obsolete
0  0x100a4847c  __assert_rtn + 72
1  0x1009a5778  ld::OpcodeFixupsEncoder::writeFixupsLinkEditContent(std::__1::span<unsigned char, 18446744073709551615ul>) + 1048
2  0x1009f4c04  ld::LayoutExecutable::writeToFile(char const*) + 8272
3  0x100a075d4  ld::Linker::run() + 8424
4  0x100993768  main + 1556
ld: Assertion failed: (((Header*)(fileOutputBuffer))->hasMachOMagic()), function writeFixupsLinkEditContent, file Layout.cpp, line 809.
make[3]: *** [memcheck-arm64-darwin] Error 1
make[2]: *** [all-recursive] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```

#### What went wrong?

Some assertion failed in the make step, I believe during linking:
```
0  0x100a4847c  __assert_rtn + 72
1  0x1009a5778  ld::OpcodeFixupsEncoder::writeFixupsLinkEditContent(std::__1::span<unsigned char, 18446744073709551615ul>) + 1048
2  0x1009f4c04  ld::LayoutExecutable::writeToFile(char const*) + 8272
3  0x100a075d4  ld::Linker::run() + 8424
4  0x100993768  main + 1556
ld: Assertion failed: (((Header*)(fileOutputBuffer))->hasMachOMagic()), function writeFixupsLinkEditContent, file Layout.cpp, line 809.
```

### Information

 * macOS architecture (`uname -m`): arm64
 * macOS version (`sw_vers`): 14.8.3
 * Xcode version (`xcrun --sdk macosx --show-sdk-version`): 15.2

[config.log](https://github.com/user-attachments/files/24778981/config.log)
[03.make.cc.log](https://github.com/user-attachments/files/24778982/03.make.cc.log)
[03.make.log](https://github.com/user-attachments/files/24778979/03.make.log)
[02.configure.cc.log](https://github.com/user-attachments/files/24778977/02.configure.cc.log)
[02.configure.log](https://github.com/user-attachments/files/24778980/02.configure.log)
[01.autogen.sh.log](https://github.com/user-attachments/files/24778978/01.autogen.sh.log)
[00.options.out.txt](https://github.com/user-attachments/files/24779008/00.options.out.txt)

--- a/coregrind/link_tool_exe_darwin.in
+++ b/coregrind/link_tool_exe_darwin.in
@@ -163,7 +163,7 @@ if ($archstr eq "arm64") {
 
 # Xcode 16 is completely borked and crashes when you use -segaddr so we rollback to
 # the old linker in the meantime
-if (@DARWIN_VERS@ >= 150000) {
+if (@DARWIN_VERS@ >= 140000) {
     $cmd = "$cmd -ld_classic";
 
 # If we're building with clang (viz, the C compiler as specified

